### ISSUE TITLE
LAMB and gradient clipping (instructions vs api)

### ISSUE THREAD

**Issue by ggaemo** [NVIDIA/apex#855]
https://nvidia.github.io/apex/advanced.html?highlight=accumulate#gradient-clipping

The instructions above show how to use pytorch's gradient clipping api for APEX.

However, when using the Fused LAMB optimizer, there is an argument 'max_grad_norm' it suggests that the Fused LAMB optimizer implements gradient clipping inside.

So is there any need for implementing gradient clipping as mentioned in the instructions in the link?

Thank you

**Comment by npuichigo** [comment]
Any update?

**Comment by Link-Piggy** [comment]
I think there is no need to implement grad clipping when using the LAMB optimizer. As the LAMB formula showing, what the algorithm does is actually implementing grad clipping by norm based on Adam.
